src/components/Launch/LaunchWorkflowForm/BlobInput.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+    FormControl,
+    FormHelperText,
+    InputLabel,
+    MenuItem,
+    Select,
+    TextField,
+    Typography
+} from '@material-ui/core';
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import { isObject } from 'lodash';
+import { BlobDimensionality } from 'models';
+import * as React from 'react';
+import {
+    blobFormatHelperText,
+    blobUriHelperText,
+    defaultBlobValue
+} from './constants';
+import { BlobValue, InputProps } from './types';
+import { getLaunchInputId } from './utils';
+
+const useStyles = makeStyles((theme: Theme) => ({
+    dimensionalityInput: {
+        flex: '1 1 auto',
+        marginLeft: theme.spacing(2)
+    },
+    formatInput: {
+        flex: '1 1 auto'
+    },
+    inputContainer: {
+        borderLeft: `1px solid ${theme.palette.divider}`,
+        marginTop: theme.spacing(1),
+        paddingLeft: theme.spacing(1)
+    },
+    metadataContainer: {
+        display: 'flex',
+        marginTop: theme.spacing(1),
+        width: '100%'
+    }
+}));
+
+/** A micro form for entering the values related to a Blob Literal */
+export const BlobInput: React.FC<InputProps> = props => {
+    const styles = useStyles();
+    const { error, label, name, onChange, value: propValue } = props;
+    const blobValue = isObject(propValue)
+        ? (propValue as BlobValue)
+        : defaultBlobValue;
```

Comment:
Suggestion to create a typeguard for `BlobValue` so we can avoid type casting. Something like:

```tsx
export function isBlobValue(data: unknown): data is BlobValue {
    return isObject(data);
}
```

And these lines would become:

```suggestion
    const blobValue = isBlobValue(propValue) ? propValue : defaultBlobValue;
```